crates/yttrium/src/sign/mod.rs

```diff
@@ -667,110 +653,615 @@ impl Client {
-            return Err(RequestError::InvalidAuth);
+        let (response_tx, response_rx) = tokio::sync::oneshot::channel();
+        self.request_tx.send((params, response_tx)).map_err(|e| {
+            RequestError::Internal(format!(
+                "Failed to send request, request_tx closed: {e}"
+            ))
+        })?;
+        let response = match response_rx.await {
+            Ok(Ok(response)) => response,
+            Ok(Err(e)) => {
+                return Err(RequestError::Internal(format!(
+                    "Failed to receive response: {e}"
+                )));
+            }
+            Err(e) => {
+                return Err(RequestError::Internal(format!(
+                    "Failed to receive response (2): {e}"
+                )));
+            }
+        };
+
+        match response {
+            Response::Success(response) => {
+                Ok(serde_json::from_value(response.result).map_err(|e| {
+                    RequestError::Internal(format!(
+                        "Failed to parse response result: {e}"
+                    ))
+                })?)
+            }
+            Response::Error(response) => Err(RequestError::Internal(format!(
+                "RPC error: {:?}",
+                response.error
+            ))),
         }
+    }
+}
 
-        let key = self
-            .key
-            .as_ref()
-            .ok_or(RequestError::Internal("Key not set".to_string()))?;
-        let url = {
-            let encoder = &data_encoding::BASE64URL_NOPAD;
-            let claims = {
-                let data = JwtBasicClaims {
-                    iss: DecodedClientId::from_key(&key.verifying_key()).into(),
-                    sub: "http://example.com".to_owned(),
-                    aud: self.relay_url.clone(),
-                    iat: crate::time::SystemTime::now()
-                        .duration_since(crate::time::UNIX_EPOCH)
-                        .unwrap()
-                        .as_secs() as i64,
-                    exp: Some(
-                        crate::time::SystemTime::now()
-                            .duration_since(crate::time::UNIX_EPOCH)
-                            .unwrap()
-                            .as_secs() as i64
-                            + 60 * 60,
-                    ),
-                };
+#[derive(Debug, PartialEq)]
+enum IncomingMessage {
+    Close(CloseReason),
+    Message(Payload),
+}
 
-                encoder.encode(serde_json::to_string(&data).unwrap().as_bytes())
-            };
-            let header = encoder.encode(
-                serde_json::to_string(&JwtHeader::default())
+#[derive(Debug, PartialEq)]
+enum CloseReason {
+    InvalidAuth,
+    Error(String),
+}
+
+#[cfg(target_arch = "wasm32")]
+type ConnectWebSocket = web_sys::WebSocket;
+
+#[cfg(not(target_arch = "wasm32"))]
+type ConnectWebSocket = tokio::sync::mpsc::UnboundedSender<String>;
+
+async fn connect(
```

Comment:
Can we rename to connect_socket or sth similar? Will be confusing with the protocol method connect for a dapp